components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -3130,6 +3131,27 @@ private void markConfidentialPropertiesUsingMetadata(IdentityProvider identityPr
                     }
                 });
         });
+
+        Map<String, List<String>> metaProvisioningConfigMap = createProvisioningConfidentialPropsMap();
+        Arrays.asList(identityProvider.getProvisioningConnectorConfigs()).forEach(provisioningConfig -> {
+            List<String> secretProperties = metaProvisioningConfigMap.get(provisioningConfig.getName());
```

Comment:
how are we differentiating the confidential properties?

Reply:
The `createProvisioningConfidentialPropsMap` method returns a map of confidential property names for each connector. This map is created based on the metadata added for each connector at OSGI level.